src/query/config/m3query-local-etcd.yml

```diff
@@ -50,3 +50,10 @@ clusters:
       backgroundHealthCheckFailLimit: 4
       backgroundHealthCheckFailThrottleFactor: 0.5
 
+readWorkerPoolPolicy:
+  grow: true
```

Comment:
Consider setting grow to false if this is only used for Dev/local testing; grow is unbounded and can lead to the issue you're trying to avoid here by spawning lots of workers.

Local load under prom is pretty light so 10 wokers should plenty